apps/aecore/src/aec_conductor.erl

```diff
@@ -1339,26 +1339,29 @@ hc_create_block_fun(ConsensusModule, TopHash) ->
     end.
 
 hc_create_block(ConsensusModule, TopHash0, Leader) ->
-    TopHash = hc_create_microblock(ConsensusModule, TopHash0, Leader),
+    VoteResult = ConsensusModule:vote_result(),
+    TopHash = hc_create_microblock(ConsensusModule, TopHash0, Leader, VoteResult),
     Res = aec_block_key_candidate:create(TopHash, Leader, Leader),
     Res.
 
-hc_create_microblock(ConsensusModule, TopHash, Leader) ->
-    case aec_block_micro_candidate:create(TopHash) of
+hc_create_microblock(ConsensusModule, TopHash, Leader, VoteResult) ->
+    CreateResult = case VoteResult of
+                        {ok, VoteTx} ->
+                            Tx = aetx_sign:tx(VoteTx),
+                                case aec_chain:get_block(TopHash) of
+                                    {ok, Block} ->
+                                        Gas = aetx:gas_limit(Tx, aec_blocks:height(Block), aec_blocks:version(Block)),
```

Comment:
Just make sure that `Gas` is reasonably large here, there was a TODO w.r.t. gas/fee?